src/Modules/ModuleTypes.jl

```diff
@@ -140,30 +140,34 @@ end
 # The same could be done rather on the level of vectors, that might be preferable if 
 # performance is ok.
 #
-mutable struct ModuleGens{T} # T is the type of the elements of the ground ring.
+@attributes mutable struct ModuleGens{T} # T is the type of the elements of the ground ring.
   O::Vector{FreeModElem{T}}
   S::Singular.smodule
   F::FreeMod{T}
   SF::Singular.FreeMod
 
-  # ModuleGens from an Array of Oscar free module elements, specifying the free module 
-  # and Singular free module, only useful indirectly
-  function ModuleGens{T}(O::Vector{<:FreeModElem}, F::FreeMod{T}, SF::Singular.FreeMod) where {T}
+  isGB::Bool
```

Comment:
Maybe like this to match the naming of the other field?
```suggestion
  is_GB::Bool
```

Reply:
`isGB` is consistent with usage in other places in code (and in Singular.jl). I think this should be changed globally. (?)

Reply:
Fair enough (not important anyway)